gcloud/monitoring/group.py

```diff
@@ -0,0 +1,502 @@
+
+    :type project: string
+    :param project: The project associated with the group.
+
+    :type group_id: string
+    :param group_id: The group ID.
+
+    :rtype: string
+    :returns: The fully qualified name of the group.
+    """
+    return 'projects/{project}/groups/{group_id}'.format(
+        project=project, group_id=group_id)
+
+
+class Group(object):
+    """A dynamic collection of monitored resources.
+
+    :type client: :class:`gcloud.monitoring.client.Client`
+    :param client: A client for operating on the metric descriptor.
+
+    :type group_id: string or None
+    :param group_id: The ID of the group.
+
+    :type display_name: string or None
+    :param display_name:
+        A user-assigned name for this group, used only for display purposes.
+
+    :type parent_id: string or None
+    :param parent_id:
+        The ID of the group's parent, if it has one.
+
+    :type filter_string: string or None
+    :param filter_string:
+        The filter string used to determine which monitored resources belong to
+        this group.
+
+    :type is_cluster: boolean
+    :param is_cluster:
+        If true, the members of this group are considered to be a cluster. The
+        system can perform additional analysis on groups that are clusters.
+    """
+    def __init__(self, client, group_id=None, display_name=None,
+                 parent_id=None, filter_string=None, is_cluster=False):
+        self.client = client
+        self._id = group_id
+        self.display_name = display_name
+        self.parent_id = parent_id
+        self.filter = filter_string
+        self.is_cluster = is_cluster
+
+        if group_id:
+            self._name = group_name_from_id(client.project, group_id)
+        else:
+            self._name = None
+
+    @property
+    def id(self):
+        """Returns the group ID.
+
+        :rtype: str or None
+        :returns: the ID of the group based on it's name.
+        """
+        return self._id
+
+    @property
+    def name(self):
+        """Returns the fully qualified name of the group.
+
+        :rtype: str or None
+        :returns:
+            The fully qualified name of the group in the format
+            "projects/<project>/groups/<id>".
+        """
+        return self._name
+
+    @property
+    def parent_name(self):
+        """Returns the fully qualified name of the parent group.
+
+        :rtype: str or None
+        :returns:
+            The fully qualified name of the parent group.
+        """
+        if not self.parent_id:
+            return None
+        return group_name_from_id(self.client.project, self.parent_id)
+
+    @property
+    def path(self):
+        """URL path to this group.
+
+        :rtype: str
+        :returns: the path based on project and group name.
+
+        :raises: :exc:`ValueError` if :attr:`name` is not specified.
+        """
+        if not self.id:
+            raise ValueError('Cannot determine path without group ID.')
```

Comment:
The path does not exist for a group without an ID - this is True for a group being created. Asking for the path of such a group should throw an error.